### ISSUE TITLE
Symfony 4 support

### ISSUE THREAD

**Issue by kor3k** [symfony/assetic-bundle#453]
please make AsseticBundle compatible with Symfony 4

related to #445 ?

**Comment by filipefernandes007** [comment]
A framework like symfony should not be so dependent on js packages. By the way, js World changes só much - require js, then gulp, then Grunt, now webpack, tomorrow another trend.

Assetic bundle was a good exemple on how to handle frontend Assets. Please, make assetic bundle compatible with sf4. Thanks in advance.

**Comment by kor3k** [comment]
@filipefernandes007 exactly. symfony now uses Webpack Encore, which is cool, but it **needs nodejs**. kinda overkill imho.

AsseticBundle rocks.

**Comment by MicWit** [comment]
This should be marked as urgent, and the documentation should have a 4.0 version as well, even if assetic is just marked as an alternative that doesn't use node.js. Most standard re-seller hosting uses Apache and doesn't have an option for node.js, and it is terrible practice to be running node.js in a development environment and Apache in production. For high end companies that run their own server this may not be a problem, but for most this is a big problem and will loose a lot of users in time.

**Comment by FelixKlauke** [comment]
There should be a way to handle assets easily, without any foreign language that doesn't blow a bunch of new files and complexity into your clean code base. Asset Bundle always was my favourite as the js world doesn't seem to be reliable even if its handled as state of the art. I don't want to install one more big environment, I don't want further commands and I don't want one more package manager or one more dependency manager. These make CI much more complicated and the project grows for nothing.

**Comment by BernhardWebstudio** [comment]
Is there anyone with a working fork willing to continue this project? I support the statements made above and am looking for a way to handle the assets as in Symfony 2+. As this repository is officially abandoned, I do not think that there is any other way than to make the bundle third-third party...

**Comment by MicWit** [comment]
Is there anyone around who worked on assetic? How hard would it be to update for symfony 4? I think with the new folder structure there would need to be a symfony 4 fork (and keep the current for 3.4) would it be worth creating an assetic2 project? Im guessing this wouldn't be supported by symfony officially so would need to be under a different account?

**Comment by kor3k** [comment]
either official or [FriendsOfSymfony](https://github.com/friendsofsymfony) should adopt it

**Comment by MicWit** [comment]
Agree, even if they consider it an alternative option (that is pure php) and just do fixes/compatibility updates.

**Comment by xabbuh** [comment]
AsseticBundle is not maintained anymore. If you really want to use it with Symfony 4, I don't see any other solution than forking and maintaining the bundle yourself. I also do not see anyone from the FriendsOfSymfony organisation maintaining the bundle in the future.

**Comment by MicWit** [comment]
Surely they will come up with some option that only requires php!

**Comment by stof** [comment]
@MicWit the thing is, none of the modern asset tooling is written in PHP. If you look at Assetic, most filters are actually using the Process component to call tools written in a different language (Node, Java, Ruby, ...).
And Assetic itself is not rally maintained anymore (@kriswallsmith stopped maintaining it years ago. I took over during some time, but I'm not using Assetic anymore since a long time either and I'm convinced using the proper frontend tooling is a better idea).

**Comment by MicWit** [comment]
@stof I used scssphp and jsqueeze, so all php. 

One of the reasons for going this way is I worked for a company (university) as a developer, which had a unix team that would only support apache, php and oracle. Because neither of the unix team members had used jsnode, they were not willing to support it.

I now develop on my own, but host o cpanel servers, so jsnode is still not an option on them either. I know in my case it would be installing jsnode on my local development machine, doing the deploy and then uploading, however, my connection is too slow (get less than 100 KB/s upload) so I generally do the deployment after uploading. 

So for those developing in Symfony, they have a few options:
1. Install nodejs locally and change the way deployments are done
2. Create my own asset deployment method
3. Find a framework that keeps to the server side language of the framework for everything
4. Develop in Symfony 3.8 permanently until a PHP solution is found for Symfony 4
5. Don't use asset management

Now when I worked for the university, the solution would have been 3, 4 or 5 (most likely 4 for a while and then 3 if no solution came up). 

In my current situation there are different sites with different requirements. Small sites probably solution 5 as there arent many/large assets and using things like .min.js files for jquery etc would be ok. 

For some more resource intensive sites (like an accounting software package I may have to write) this would not be good enough. I wouldn't want to use solution 4 for security reasons (and a few other reasons), solutions 1-3 are the only options, in this case probably 1 or 2 although both are going to be a pain and take time I dont really have to implement. 

There are going to be a lot of other businesses out there in similar circumstances, and I'm guessing a lot will go for option 3 if they see Symfony isn't going to suit their needs. 

This is not good for the Symfony community as a whole. Giving an option of assetic (even a cut down version of assetic just to compress css and js on deployment) would be beneficial to the community. I know that assetic has not been under development for a while, but couldnt there be a fork that was kept just for Symfony 4 compatibility? 

Btw @stof, having worked on it, how hard in your opinion do you think it would be to make compatible with Symfony 4?

**Comment by FelixKlauke** [comment]
Seems like the problem is not the question, if a project like this has to exist, but who would take care about that it exists. Any volunteers?

**Comment by sanpii** [comment]
If you want I updated assetic and this bundle for php 7.1 and symfony 4 : https://github.com/sanpii/assetic-bundle

**Comment by BernhardWebstudio** [comment]
@xabbuh nice to hear that AsseticBundle is not deprecated, but if so, could you please update the description of this repository? Or did I missunderstand something? 
![screen shot assetic deprecation](https://user-images.githubusercontent.com/8596965/43138618-bd019c40-8f4f-11e8-8917-3ea7e9b2000e.png)

**Comment by xabbuh** [comment]
@BernhardWebstudio Oops, I am sorry. I made a typo in my comment. It should have been "now" and not "not" (updated above).

**Comment by flolivaud** [comment]
@sanpii Your bundle don't work with symfony 4 (requirements are in symfony ~3.0).

**Comment by sanpii** [comment]
@flolivaud Fixed. I also allowed issues on my fork, if I missed something else.

**Comment by sweiguny** [comment]
@sanpii your master branch of assetic-bundle seems to still not support Symfony4, despite your assetic project does. are you going to merge the two branches?
or how can i load your symfony4 branch of assetic-bundle?

**Comment by multifinger** [comment]
WTF?! assetic was excelent in his simplisity. 
webpack casuses a lot of pain in installing dependensies (try to import jquery-mobile for example!)

**Comment by lhadjadj** [comment]
I agree with you. 
If I want to make JS dor Front-End, I use Angular and not PHP + Webpack + "Encore". My site use symfony 3.4 and I would like to upgrade to version 4 or 5. The only problem is Sensio's discontinuation of Assetic for Webpack + "Encore".
 I haven't problems with nodejs or yarn, I use it every day with Angular+Springboot applications. 
After two days of hard work to make webpack work on Symfony, only on the first page (lol, I have jquery, Zurb foundation and many JS) I'am stopping, webpack + "Encore" is not at all, practical, I don't control the loading, I have big files (3 mo) and I have) lot of  bugs. 
So I'm very happy, if there's a fork of Assetic. If I can help, I would be happy to.

Translated with www.DeepL.com/Translator (free version)
